### ISSUE TITLE
How to change the completion sources based on line contents

### ISSUE THREAD

**Issue by snoblenet** [hrsh7th/nvim-cmp#1889]
When a line matches one of these two patterns...
- `const XXX = require('`
- `import XXX from '`

...I'd like to do one of these two things:
- make `cmp_paths` the top-most source (best the option)
- make `cmp_paths` the only source (probably a satisfactory alternative)

Is this possible? I could not find any clues in the docs, but I might have missed something.

Thanks :-)

**Comment by xzbdmw** [comment]
Yeah, writing a custom comparator and based on your line content to change weight.
